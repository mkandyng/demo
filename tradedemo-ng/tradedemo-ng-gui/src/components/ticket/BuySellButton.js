import React from "react";

export default function BuySellButton(props) {
    return (
        <div id={props.containerId} onClick={props.handleOnClick}>
             <label>{props.label}</label>
             <button id={props.id}>{props.buttonName}</button>
       </div>
   )
}
