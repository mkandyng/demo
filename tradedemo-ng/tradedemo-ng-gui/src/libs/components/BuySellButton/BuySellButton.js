import React from "react";
import PropTypes from 'prop-types';

/**
 * [BuySellButton Buy and Sell buttons with Label displaying buy/sell prices]
 */
export default function BuySellButton(props) {
    const { id,
            containerId,
            label,
            buttonName,
            handleOnClick } = props;
    return (
        <div id={containerId} onClick={handleOnClick}>
               <label>{label}</label>
               <button id={id}>{buttonName}</button>
         </div>
   )
}

BuySellButton.propTypes = {
    id: PropTypes.string.isRequired,
    containerId: PropTypes.string.isRequired,
    label: PropTypes.string,
    buttonName: PropTypes.string.isRequired,
    handleOnClick: PropTypes.func.isRequired
};
