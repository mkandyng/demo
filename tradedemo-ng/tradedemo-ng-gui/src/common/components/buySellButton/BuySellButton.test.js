import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import BuySellButton from "./BuySellButton";

/**
 * This is unit test for the BuySellButton widget
 * It leverage snapshot to test the presentation
 * and a functional test to check the onClick call
 */

describe("BuySellButton", () => {
  const props = {
          id: "id",
          containerId: "divId",
          label: "label",
          buttonName: "buttonName",
          handleOnClick: jest.fn()};

    it("should render component comparing with previous snapshot shallow markup", () => {
        // Given

        // When
        const component = shallow(<BuySellButton {...props} />);

        // Then
        expect(toJson(component)).toMatchSnapshot();
    });

    it("should call handleOnClick prop with given event", () => {
        // Given
        const event = 1
        const component = shallow(<BuySellButton {...props} />);

        // When
        component.find("div#" + props.containerId).simulate('click', event);

        // Then
        expect(props.handleOnClick).toBeCalledWith(event);
    });

});
