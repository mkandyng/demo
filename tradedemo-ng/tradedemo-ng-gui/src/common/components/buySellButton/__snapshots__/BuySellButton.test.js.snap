// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuySellButton should render component comparing with previous snapshot shallow markup 1`] = `
<div
  id="divId"
  onClick={[MockFunction]}
>
  <label>
    label
  </label>
  <button
    id="id"
  >
    buttonName
  </button>
</div>
`;
